zebra-rpc/src/methods.rs

```diff
@@ -248,6 +262,33 @@ where
         }
         .boxed()
     }
+
+    fn get_best_block_hash(&self) -> BoxFuture<Result<GetBestBlockHash>> {
+        let mut state = self.state.clone();
+
+        async move {
+            let request = zebra_state::Request::Tip;
+            let response = state
+                .ready()
+                .and_then(|service| service.call(request))
+                .await
+                .map_err(|error| Error {
+                    code: ErrorCode::ServerError(0),
+                    message: error.to_string(),
+                    data: None,
+                })?;
+
+            match response {
+                zebra_state::Response::Tip(Some((_height, hash))) => Ok(GetBestBlockHash(hash)),
+                _ => Err(Error {
+                    code: ErrorCode::ServerError(0),
+                    message: "Block not found".to_string(),
+                    data: None,
+                }),
```

Comment:
There are two different cases here:
1. Bug?: `Tip(None) =>`
    What does lightwalletd expect if the state is empty?
    Can you add a unit test for this case?

2. Minor: `_ =>`
    I think we should `panic!` here if the `zebra-state::Response` is not the right variant.
    Like https://github.com/ZcashFoundation/zebra/pull/3707#discussion_r820150235

Reply:
Ok with the panic, was added at https://github.com/ZcashFoundation/zebra/pull/3754/commits/2db813b7bd858a0c953008116eabd0f41186e102

I dont understand how to do the test you want.

Reply:
> I dont understand how to do the test you want.

Thinking about this a bit more, I'm not sure if a unit test is going to be useful.

So I'd like an integration test, but I think our existing integration test covers that.